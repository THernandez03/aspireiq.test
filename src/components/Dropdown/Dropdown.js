import React, { useState, useEffect, useCallback } from "react";

import { debounce } from "../../utils";
import styles from "./Dropdown.module.scss";

export const Dropdown = ({ term, hiddenValues, onItemClick = () => {} }) => {
  const [filteredValues, setFilteredValues] = useState([]);

  const fetchEmails = async (term) => {
    const response = await fetch(`/api/${term}`);
    const { data } = await response.json();
    setFilteredValues(data);
  };

  const debouncedFetch = useCallback(debounce(fetchEmails, 500), []);

  useEffect(() => {
    if (!term) {
      return setFilteredValues([]);
    }

    debouncedFetch(term);
  }, [term]);

  if (!filteredValues.length) return null;

  return (
    <div className={styles.wrapper}>
      {filteredValues
        .filter((value) => !hiddenValues.has(value))
        .map((value) => (
          <span key={value} className={styles.item} onClick={onItemClick}>
            {value}
          </span>
        ))}
    </div>
  );
};
