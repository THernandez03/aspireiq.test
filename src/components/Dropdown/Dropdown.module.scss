.wrapper{
  position: absolute;
  display: flex;
  flex-direction: column;
  top: 55px;
  background: #FFFFFF;
  box-shadow: 1px 1px 10px 0px #d7d7d7;
  overflow: auto;
  height: auto;
  max-height: 250px;

  > .item{
    padding: 10px;

    &:hover{
      background: #EFF5F9;
    }
  }
}
