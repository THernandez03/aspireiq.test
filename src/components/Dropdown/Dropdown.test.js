import React from "react";
import { render } from "@testing-library/react";

import { Dropdown } from "./Dropdown";

describe("<Dropdown/>", () => {
  it("should render a Dropdown", () => {
    const { container } = render(<Dropdown />);
    expect(container).toMatchSnapshot();
  });
});
