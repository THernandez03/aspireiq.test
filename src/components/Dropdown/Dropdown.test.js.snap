// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown/> should render a Dropdown 1`] = `<div />`;
